nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -118,7 +118,6 @@ class SimpleActionServer
 
   void deactivate()
   {
-    std::lock_guard<std::mutex> lock_goal_handle(update_mutex_);
```

Comment:
Why are we removing the lock?

Reply:
The lock called in `terminate_goals` would hang since we are locking the mutex twice in same function call. I've re-added the lock but now using a `std::recursive_mutex`